backend/rdf4j/src/main/java/org/dotwebstack/framework/backend/rdf4j/query/SubjectQueryBuilder.java

```diff
@@ -10,23 +17,76 @@
 
 class SubjectQueryBuilder extends AbstractQueryBuilder<SelectQuery> {
 
-  private SubjectQueryBuilder(QueryEnvironment environment) {
+  private static final Integer DEFAULT_PAGE = 1;
+
+  private static final Integer DEFAULT_PAGESIZE = 10;
+
+  private static final Variable SUBJECT_VAR = SparqlBuilder.var("s");
+
+  private final JexlEngine jexlEngine;
+
+  private final NodeShape nodeShape;
+
+  private SubjectQueryBuilder(final QueryEnvironment environment, final JexlEngine jexlEngine) {
     super(environment, Queries.SELECT());
+    this.jexlEngine = jexlEngine;
+    this.nodeShape = this.environment.getNodeShapeRegistry().get(this.environment.getObjectType());
   }
 
-  static SubjectQueryBuilder create(QueryEnvironment environment) {
-    return new SubjectQueryBuilder(environment);
+  static SubjectQueryBuilder create(final QueryEnvironment environment,
+                                    final JexlEngine jexlEngine) {
+    return new SubjectQueryBuilder(environment, jexlEngine);
+  }
+
+  String getQueryString(final Map<String, Object> arguments,
+                        final GraphQLDirective sparqlDirective) {
+    arguments.putIfAbsent("page", DEFAULT_PAGE);
+    arguments.putIfAbsent("pageSize", DEFAULT_PAGESIZE);
+    final MapContext context = new MapContext(arguments);
+
+    this.query.select(SUBJECT_VAR)
+    .where(GraphPatterns
+      .tp(SUBJECT_VAR, ns(RDF.TYPE), ns(this.nodeShape.getTargetClass())))
+      .limit(getLimitFromContext(context, sparqlDirective))
+      .offset(getOffsetFromContext(context, sparqlDirective));
+
+    return this.query.getQueryString();
   }
 
-  String getQueryString() {
-    Variable subjectVar = SparqlBuilder.var("s");
-    NodeShape nodeShape = environment.getNodeShapeRegistry().get(environment.getObjectType());
+  Integer getLimitFromContext(MapContext context, GraphQLDirective sparqlDirective) {
+    JexlExpression limitExpression = this.jexlEngine.createExpression(DirectiveUtils
+            .getStringArgument(Rdf4jDirectives.SPARQL_ARG_LIMIT, sparqlDirective));
+    Object limitObject = limitExpression.evaluate(context);
+
+    if (!(limitObject instanceof Integer)) {
+      throw new IllegalArgumentException(("The given limit expression is invalid"));
+    }
+
+    Integer limit = (Integer) limitObject;
+
+    if (limit < 1) {
+      throw new IllegalArgumentException("The given pageSize is invalid");
+    }
+
+    return limit;
+  }
+
+  Integer getOffsetFromContext(MapContext context, GraphQLDirective sparqlDirective) {
```

Comment:
is er een reden om geen access modifiers aan de class en methods toe te voegen?
```suggestion
  protected Integer getOffsetFromContext(MapContext context, GraphQLDirective sparqlDirective) {
```

Reply:
Geen specifieke reden, maar was meer omdat ik het in lijn heb gehouden met hoe de class en de applicatie was opgezet. Ben het wel met je eens, we hebben die visibility keywords niet voor niets gekregen, verbositeit lijkt me in dit geval wel het meest wenselijk, dus ook protected uitschrijven, weet niet hoe de rest hier over denkt, maar lijkt me een goede voor de code conventies 👍